pkg/didcomm/transport/http/inbound.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// MessageHandler is a function that handles the inbound request payload
+// the payload will be unpacked prior to calling this function.
+// this function will be called for all incoming requests
+// and must trigger outbound requests to resume DID-Exchange flow
+type MessageHandler func(payload []byte)
+
+// NewInboundHandler will create a new handler to enforce Did-Comm HTTP transport specs
+// then routes processing to the mandatory 'msgHandler' argument.
+//
+// Arguments:
+// * 'msgHandler' is the handler function that will be executed with the inbound request payload.
+//    Users of this library must manage the handling of all inbound payloads in this function.
+func NewInboundHandler(msgHandler MessageHandler) (http.Handler, error) {
+	if msgHandler == nil {
+		log.Println("Error creating a new inbound handler: message handler function is nil")
+		return nil, errors.New("Failed to create NewInboundHandler")
+	}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		processPOSTRequest(w, r, msgHandler)
+	}), nil
+}
+
+// TODO Log error message with a common logger library for aries-framework-go
+func processPOSTRequest(w http.ResponseWriter, r *http.Request, messageHandler MessageHandler) {
+	if valid := validateHTTPMethod(w, r); !valid {
+		return
+	}
+	if valid := validatePayload(r, w); !valid {
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error reading request body: %s - returning Code: %d", err, http.StatusInternalServerError)
+		http.Error(w, "Failed to read payload", http.StatusInternalServerError)
+	}
+
+	// TODO add Unpack(body) call here
```

Comment:
Does Unpack happen within the transport code?

Reply:
@troyronda based on our discussion, I assume the answer is "yes". My thoughts are that unpacking has to be done across all DIDComm messages, so the transport layer seems like a natural choice. This decision can be revisited later on.

Reply:
technically we can call Pack/Unpack at DidComm (upper level),  but at the transport level, it will be handled in one place for all calls.